### PR TITLE
[UIKit] Update to Xcode8-GM

### PR DESCRIPTION
**Pull request by dalexsoto**

### CODE REVIEW

File: src/uikit.cs [hunk 1]
```diff
@@ -127,6 +143,43 @@ interface NSAttributedStringAttachmentConveniences {
...
+	[Abstract] // abstract class that should not be used directly
+	[BaseType (typeof (NSObject))]
+	interface UIFeedbackGenerator {
+
+		[Export ("prepare")]
+		void Prepare ();
+	}
+
+	[NoWatch, NoTV, iOS (10,0)]
+	[BaseType (typeof(UIFeedbackGenerator))]
```
Comment (spouliot) [inline src/uikit.cs]: in doubt (no designated initializer) let's `[DisableDefaultCtor]`

File: src/uikit.cs [hunk 2]
```diff
@@ -127,6 +143,43 @@ interface NSAttributedStringAttachmentConveniences {
...
+		[Export ("initWithStyle:")]
+		IntPtr Constructor (UIImpactFeedbackStyle style);
+
+		[Export ("impactOccurred")]
+		void ImpactOccurred ();
+	}
+
+	[NoWatch, NoTV, iOS (10,0)]
+	[BaseType (typeof(UIFeedbackGenerator))]
+	interface UINotificationFeedbackGenerator {
```
Comment (spouliot) [inline src/uikit.cs]: this likely needs to inline the base .ctor - we only warn for designated initializer (but I guess it should be one)

File: src/uikit.cs [hunk 3]
```diff
@@ -127,6 +143,43 @@ interface NSAttributedStringAttachmentConveniences {
...
+
+		[Export ("notificationOccurred:")]
+		void NotificationOccurred (UINotificationFeedbackType notificationType);
+	}
+
+	[NoWatch, NoTV, iOS (10,0)]
+	[BaseType (typeof(UIFeedbackGenerator))]
+	interface UISelectionFeedbackGenerator {
```
Comment (spouliot) [inline src/uikit.cs]: same (base .ctor inline)

File: src/uikit.cs (hunk 2 above)
Comment (spouliot) [inline src/uikit.cs]: also not clear if we should remove the default .ctor ... did you check if any doc/samples were added ?

Comment (dalexsoto) [inline src/uikit.cs]: mmm there is not `initWithxxx:` in the base ctor and docs do not have any info about it

File: src/uikit.cs (hunk 3 above)
Comment (dalexsoto) [inline src/uikit.cs]: mmm there is not initWithxxx: in the base ctor and docs do not have any info about it

File: src/uikit.cs [hunk 4]
```diff
@@ -127,6 +143,44 @@ interface NSAttributedStringAttachmentConveniences {
...
+		[Export ("initWithStyle:")]
+		IntPtr Constructor (UIImpactFeedbackStyle style);
+
+		[Export ("impactOccurred")]
+		void ImpactOccurred ();
+	}
+
+	[NoWatch, NoTV, iOS (10,0)]
+	[BaseType (typeof (UIFeedbackGenerator))]
+	interface UINotificationFeedbackGenerator {
```
Comment (dalexsoto) [inline src/uikit.cs]: @spouliot there is not initWithxxx: in the base ctor and docs do not have any info about it

File: src/uikit.cs [hunk 5]
```diff
@@ -127,6 +143,44 @@ interface NSAttributedStringAttachmentConveniences {
...
+
+		[Export ("notificationOccurred:")]
+		void NotificationOccurred (UINotificationFeedbackType notificationType);
+	}
+
+	[NoWatch, NoTV, iOS (10,0)]
+	[BaseType (typeof (UIFeedbackGenerator))]
+	interface UISelectionFeedbackGenerator {
```
Comment (dalexsoto) [inline src/uikit.cs]: @spouliot there is not initWithxxx: in the base ctor and docs do not have any info about it

File: src/uikit.cs (hunk 4 above)
Comment (dalexsoto) [inline src/uikit.cs]: Just tested an ObjC app and I am able to get a handle from the default ctor

File: src/uikit.cs (hunk 5 above)
Comment (dalexsoto) [inline src/uikit.cs]: Just tested an ObjC app and I am able to get a handle from the default ctor

File: src/uikit.cs (hunk 4 above)
Comment (spouliot) [inline src/uikit.cs]: You're right, I misread the base type (for `UIImpactFeedbackGenerator`).

File: src/uikit.cs (hunk 5 above)
Comment (spouliot) [inline src/uikit.cs]: same :)

**Comment by spouliot** [comment]
LGTM pending bots

**Pull request merged**